hydra/core/override_parser/types.py

```diff
@@ -413,15 +420,17 @@ def _get_value_element_as_str(
         elif isinstance(value, dict):
             s = comma.join(
                 [
-                    f"{k}{colon}{Override._get_value_element_as_str(v, space_after_sep=space_after_sep)}"
+                    f"{Override._get_value_element_as_str(k)}{colon}"
+                    f"{Override._get_value_element_as_str(v, space_after_sep=space_after_sep)}"
                     for k, v in value.items()
                 ]
```

Comment:
For my sanity reading this later, can you refactor a bit?
I think at this point a regular loop with temp variables will be much easier to parse.

Reply:
Done in 4f2665eec